src/context/historyContext.jsx

```diff
@@ -0,0 +1,47 @@
+import { createContext, useContext, useState } from 'react'
+import axios from 'axios'
+import { useAuth } from 'context'
+import { POST_HISTORY } from 'utils/constants/apiEndPoints'
+import checkIfPresent from 'utils/functions/checkIfPresent'
+
+const historyContext = createContext(null)
+
+function HistoryProvider( { children } ){
+
+    const [ historyState, setHistory] = useState([])
+
+    const { authState: {isLoggedIn, token} } = useAuth()
+
+    function addToHistory( item ){
+        if (isLoggedIn && !checkIfPresent(item._id, historyState)){
+            axios.post(POST_HISTORY,
+                { video: item },
+                {
+                    headers: {
+                        authorization: token,
+                    },
+                }
+            ).then((response) => {
+                setHistory(response.data.history);
+            }).catch((error) => {
+                console.log(error)
+            })

```

Comment:
Suggestion:- It is good to use `async-await` and also `try-catch` block for API calls

Reply:
Have made the changes to make use of async-await and try-catch.